RFPStrategy/RFPMilestone/spec.ts

```diff
@@ -0,0 +1,78 @@
+import { Address, BeforeAll, Event, LiveObject, OnEvent, Property, saveAll,Spec } from '@spec.dev/core'
+
+import { getStatusFromInt } from '../../shared/status'
+
+/**
+ * RFP Milestone details
+ */
+@Spec({
+    uniqueBy: ['strategyId', 'id', 'chainId']
+})
+class RFPMilestone extends LiveObject {
+
+    @Property()
+    milestoneId: number
+
+    @Property()
+    strategyId: Address
+
+    @Property()
+    metadataProtocol: number
+
+    @Property()
+    metadataPointer: stringg
+
+    @Property()
+    status: string
+
+    // ====================
+    // =  Event Handlers  =
+    // ====================
+
+    @BeforeAll()
+    setCommonProperties(event: Event) {
+        this.strategyId = event.origin.contractAddress;
+        this.milestoneId = event.data.milestoneId;
+    }
+
+    @OnEvent('allov2.RFPSimple.MilestonesSet')
```

Comment:
For handlers that use `saveAll(...)`, I usually turn off autosaving so that spec doesn't try to save `this` at the end of the function automatically:
```typescript
@OnEvent('allov2.RFPSimple.MilestonesSet', { autoSave: false })
```